OneCommunityGlobal/HighestGoodNetworkApp
# Jianjun - add total people report feature

# Description
(PRIORITY HIGH) **Jae: Create “Teams Summary” on the Reports page**
Admin/Owner Login → Reports → Reports
In the view below, I need a “Total People Report” button to see exactly how many total people have contributed time to the project for a designated time period
People must have had at least 10 hours logged for them to count/be included
Should be able to choose the period of time and the result should show on this page or as a Modal
Should include a “Total Hours” section showing total tangible time logged by all team members during the selected period
Would like to run a “details” report that will list all the people and hours contributed by each for that time period too, with clickable names to visit their profiles
Need an “i” by the button to explain what the button does (all of the above)
This is needed because it is a metric we’re using from here on forward to measure the growth and progress of our nonprofit. So I want to be able to easily run this for individual months and/or an entire fiscal year to compare results with previous years.
A bar graph visualization of each month when seeing the yearly summary would be super helpful too

## Main changes explained:
- Add the “Total People Report” button and its 'i' data tip. (The content for the other two button is out of the scope of this PR)
- create the information card that shows the number of total qualified members (with more than 10 hours) and the total tangible time logged in this period.
- Two bar graph to show the number of the valid members monthly or yearly.
- A detail table listing all the people and the hours in the period with link to the profile, which can hide/show by clicking the button.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. Admin/Owner Login → Reports → Reports
5. try to select different start/end date and then click the 'Total People Report' button.

## Screenshots or videos of changes:
<img width="591" alt="1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/289931cf-e694-4d2a-864a-a8db5d793796">

<img width="487" alt="2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/f48f990b-38a6-48df-9955-b342412849f2">

<img width="493" alt="3" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/9314962/33a36a44-1f83-411d-892f-49166f44e48b">

## How to test:
Note that the two bar graph will not always show in this section. It depends on the length of the time period you select.
Now the conditions are as following:
The time period is less than one month: no bar graph.
between one month to about twelve month: show monthly bar graph.
The start date and the end date are in different year: show yearly bar graph.
